django_usaid_clinicfinder/clinicfinder/tasks.py

```diff
@@ -223,3 +225,59 @@ def run(self, lookuppointofinterest_id, **kwargs):
                 exc_info=True)
 
 location_finder = Location_Finder()
+
+
+class PointOfInterest_Importer(Task):
+
+    """
+    Task to take dict import the data
+    """
+    name = "clinicfinder.tasks.pointofinterest_importer"
+
+    class FailedEventRequest(Exception):
+
+        """
+        The attempted task failed because of a non-200 HTTP return
+        code.
+        """
+
+    def run(self, poidata, **kwargs):
+        """
+        Returns count of imported records
+        """
+        l = self.get_logger(**kwargs)
+
+        l.info("Processing new point of interest import data")
+        imported = 0
+        try:
+            for line in poidata:
+                if "Latitude" in line and "Longitude" in line:
+                    poi_point = Point(float(line["Longitude"]), float(line["Latitude"]))
+                    # check if point exists
+                    locations = Location.objects.filter(point=poi_point)
+                    if locations.count() == 0:
+                        # make a Location
+                        location = Location()
+                        location.point = poi_point
+                        location.save()
+                    else:
+                        # Grab the top of the stack
+                        location = locations[0]
+                    # Create new point of interest with location
+                    poi = PointOfInterest()
+                    poi.location = location
+                    poi.data = line
+                    poi.save()
+                    imported += 1
+                    l.info("Imported: %s" % line["Clinic Name"])
+                else:
+                    l.info("Row missing point data, not imported")
```

Comment:
Perhaps it would be useful to add the line number or line["Clinic Name"] to this?